test/superlogger_test.rb

```diff
@@ -64,6 +64,41 @@ def output
     assert fields.key?("request_id")
   end
 
+  # The extra fields to log are defined in `/test/dummy/config/application.rb`.
+  # Unfortunately, because `Superlogger::SuperloggerMiddleware` is initialised
+  # together with the Rails application, and because we cannot programmatically
+  # reinitialise the Rails application or run multiple instances of it, we are
+  # only able to test a single Superlogger configuration. Once the Rails
+  # application is initialised, the middleware stack cannot be changed.
+  #
+  # This effectively means that we are unable to change the Superlogger settings
+  # in the Rails application configuration on a per-test basis. For the sake of
+  # testing, we choose the Superlogger settings that let us test the presence of
+  # behaviours and set the extra fields to log.
```

Comment:
For documentation: To frame this differently, this means that we cannot test that the logging middleware works even when `Superlogger.log_extra_fields` is undefined (nil)?

Reply:
No, we can test that the logging middleware works when `Superlogger.log_extra_fields` is `nil`, just not in the same test suite run as another test case which requires that `Superlogger.log_extra_fields` is set to some other value.

Specifically, Superlogger is set to initialise itself together with Rails:
https://github.com/moexmen/superlogger/blob/97d3370885e6f5bdccfe7e93bc4607d50dec61bb/lib/superlogger/railtie.rb#L1-L7

After Superlogger's initialisation, any change made to its settings doesn't do anything as the `setup` method is only called once:
https://github.com/moexmen/superlogger/blob/97d3370885e6f5bdccfe7e93bc4607d50dec61bb/lib/superlogger.rb#L9-L15

Calling the `setup` method manually doesn't work because Rails freezes its middleware stack after initialisation. Additionally, there is no way to programmatically unload Rails to load it again with a different configuration.

It might be possible to use a gem such as [`combustion`](https://github.com/pat/combustion) to workaround the above limitations, but it will likely require overhauling the entire testing setup (including modernising it to use RSpec).